cmake/AddPdfTarget.cmake

```diff
@@ -6,32 +6,21 @@ function (add_pdf_target docname lang entities figures)
     set(BUILD_DIR "${DOCDIR_BUILD}/${lang}")
     file(MAKE_DIRECTORY "${BUILD_DIR}")
 
-    # Determine paper format depending on language (which maps to the document's directory name)
-    # * for language "C" (fallback language) determine paper format based on current locale
-    # * all others use A4.
-    set (XSLTFLAGS_FO "--stringparam paper.type A4")
-    if (lang STREQUAL "C")
-        # For the fallback language determine paper format depending on locale
-        # Only US or C will be set to letter. All others use A4.
-        set (ENV_LANG $ENV{LANG})
-        if (ENV_LANG AND ENV_LANG MATCHES ".*_us.*|C") # Replacing ENV_LANG here with if ($ENV{LANG}) won't work.
-            set (XSLTFLAGS_FO "--stringparam paper.type letter")
-        endif()
-    endif()
```

Comment:
While I like you moved this inside the stylesheet it now no longer does exactly the same as was done here.

The assumption behind the logic above is that only native English speaking countries use Legal as paper format. All others default to A4 (this may be a false assumption by the way, just explaining the rationale that was used here).
So for all languages other than English ("C"), the choice is straightforward: use A4.
"C"/English is special because we consider that the fallback language. That is, if someone natively speaking a language for which we don't have the documentation, that person is assumed to go for English. Add to that the English edition is the most complete one so the most preferred version in any case. As said before most of the world uses A4 formatting, so unless the user choosing English because said user is native English, the expectation is to get a A4 formatted document. This final subtlety is lost in your change.

Now having explained all that, I think the above logic is probably trying to be too smart also. It would be much better to take what you have written in the stylesheet as default but allow the user a way to override by providing a PDF_PAPER_FORMAT cmake option. With that a user could run
cmake -DPDF_PAPER_FORMAT=A4 <src-dir>
or
cmake -DPDF_PAPER_FORMAT=Letter
in cases our simplified algo doesn't do what the user wants.

Reply:
I see, but I'm reading cmake and ninja documents now. So it is difficult for me to fix it right now.

Reply:
It's a minor issue IMO. Can be revised later.

Reply:
> While I like you moved this inside the stylesheet it now no longer does exactly the same as was done here.

Perhaps we should ask the user's OS on initialization and then use a string-parameter to overwrite A4?.

> The assumption behind the logic above is that only native English speaking countries use Legal as paper format. All others default to A4 (this may be a false assumption by the way, just explaining the rationale that was used here).
> So for all languages other than English ("C"), the choice is straightforward: use A4.

Not exactly: in en_GB it should be A4. CA's P series is a rounding of US formats to 5 mm. And some es speaking countries in the US "backyard" use letter called „Carta“ there.

https://en.wikipedia.org/wiki/Paper_size
https://unicode-org.github.io/cldr-staging/charts/latest/supplemental/territory_information.html

Reply:
ToDo.